### PR TITLE
Development build per branch versioning

### PR DESCRIPTION
**Pull request by jmozgawa**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -173,6 +173,9 @@ jobs:
     working_directory: ~/TheBrain2.0
     steps:
       - checkout
+      - run:
+          name: Generating branch name
```
Comment (jmozgawa) [inline circle.yml]: Name `branch name` is misleading, I suggest to use something like `build version label`

File: circle.yml [hunk 2]
```diff
@@ -173,6 +173,9 @@ jobs:
...
+          command: BRANCH_NAME=$(echo ${CIRCLE_BRANCH,,} | md5sum | awk -v branch=$CIRCLE_BRANCH '{ print tolower(substr(branch,0,15))"-"substr($1,0,8) }') && echo "export BRANCH_NAME=$BRANCH_NAME" >> $BASH_ENV
```
Comment (jmozgawa) [inline circle.yml]: Env var `BRANCH_NAME` should also be renamed to something like `BUILD_VERSION_LABEL`, please rename all occurrences.

**Comment by lgandecki** [comment]
Merging it for now but please move this to a script that we can run, so we don't duplicate such a complex command twice in the circleci (and in general I would rather try to avoid such complex commands in the circle config)

**Comment by lgandecki** [comment]
@lucetius ^

**Pull request merged**